### PR TITLE
Remove lock on Pulumi commands

### PR DESCRIPTION
**Pull request by KevinFairise2**
### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by L3n41c — CHANGES_REQUESTED** [review]
`sm.stacks` is written here:
https://github.com/DataDog/datadog-agent/blob/7d39806fdf827b9d24aa4a65b9c2457b37ca75ca/test/new-e2e/pkg/utils/infra/stack_manager.go#L200

Without this `StackManager.lock` `RWMutex` anymore, what prevents this write from happening concurrently to the reads done:
https://github.com/DataDog/datadog-agent/blob/7d39806fdf827b9d24aa4a65b9c2457b37ca75ca/test/new-e2e/pkg/utils/infra/stack_manager.go#L100
https://github.com/DataDog/datadog-agent/blob/7d39806fdf827b9d24aa4a65b9c2457b37ca75ca/test/new-e2e/pkg/utils/infra/stack_manager.go#L127
https://github.com/DataDog/datadog-agent/blob/7d39806fdf827b9d24aa4a65b9c2457b37ca75ca/test/new-e2e/pkg/utils/infra/stack_manager.go#L144
https://github.com/DataDog/datadog-agent/blob/7d39806fdf827b9d24aa4a65b9c2457b37ca75ca/test/new-e2e/pkg/utils/infra/stack_manager.go#L184
https://github.com/DataDog/datadog-agent/blob/7d39806fdf827b9d24aa4a65b9c2457b37ca75ca/test/new-e2e/pkg/utils/infra/stack_manager.go#L200
https://github.com/DataDog/datadog-agent/blob/7d39806fdf827b9d24aa4a65b9c2457b37ca75ca/test/new-e2e/pkg/utils/infra/stack_manager.go#L272

[Go maps are not thread-safe](https://go.dev/blog/maps#concurrency). That’s why a lock is needed to protect their access.

This having been said, we can probably reduce the scope of the mutex lock to protect only the accesses to the `sm.stacks` map and not be held when Pulumi manipulates its stacks.

File: test/new-e2e/pkg/utils/infra/stack_manager.go [hunk 1]
```diff
@@ -121,13 +147,12 @@ func (sm *StackManager) DeleteStack(ctx context.Context, name string) error {
...
 func (sm *StackManager) ForceRemoveStackConfiguration(ctx context.Context, name string) error {
-	sm.lock.Lock()
-	defer sm.lock.Unlock()
 
-	stack, ok := sm.stacks[name]
+	storedStack, ok := sm.stacks.Get(name)
 	if !ok {
 		return fmt.Errorf("unable to remove stack %s: stack not present", name)
 	}
+	stack := storedStack
```
Comment (pducolin) [inline test/new-e2e/pkg/utils/infra/stack_manager.go:155]: **❓ question**
Why do you copy the stack reference ?

Comment (pducolin) [inline test/new-e2e/pkg/utils/infra/stack_manager.go:155]:
```suggestion
	stack, ok := sm.stacks.Get(name)
	if !ok {
		return fmt.Errorf("unable to remove stack %s: stack not present", name)
	}
```
Should be enough

**Review by pducolin — APPROVED** [review]

File: test/new-e2e/pkg/utils/infra/stack_manager.go [hunk 2]
```diff
@@ -43,10 +43,42 @@ var (
...
+func (s *safeStackMap) Set(key string, value *auto.Stack) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	s.stacks[key] = value
+}
+
+func (s *safeStackMap) Range(f func(string, *auto.Stack)) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
```
Comment (pducolin) [inline test/new-e2e/pkg/utils/infra/stack_manager.go:75]: **💬 suggestion**
As you use items inside the map without changing their reference, a read lock should be enough
```suggestion
	s.lock.RLock()
	defer s.lock.RUnlock()
```

**Review by pducolin — APPROVED** [review]

File: test/new-e2e/pkg/utils/infra/stack_manager.go (hunk 1 above)
Comment (KevinFairise2) [reply test/new-e2e/pkg/utils/infra/stack_manager.go:155]: Yes that's a leftover from previous tests I made, I though I had delete everything

**Review by L3n41c — APPROVED** [review]
I agree with @pducolin’s comments. But I’ve nothing to add.

File: test/new-e2e/pkg/utils/infra/stack_manager.go (hunk 2 above)
Comment (KevinFairise2) [reply test/new-e2e/pkg/utils/infra/stack_manager.go]: done

**Pull request merged**